internal/state/rollback.go

```diff
@@ -0,0 +1,86 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/tendermint/tendermint/version"
+)
+
+// Rollback takes a state at height n and overwrites it with the state
+// at height n - 1. Note state here refers to tendermint state not application state.
```

Comment:
Minor wording suggestions:

```suggestion
// Rollback overwrites the current Tendermint state (height n) with the most
// recent previous state (height n - 1).
// Note that this function does not affect application state.
```

Rationale: Clarifies that "height n" is a description of the current state rather than an expected parameter.

Are there other preconditions we need to document here, e.g., that the node shouldn't be running? In the case where the node has already stalled I guess this is fine—but what if someone runs this against a node that's not stalled?

Reply:
Yup good catch, we should definitely specify that the node should not be running. In the case of a divergent app hash the node usually shuts down anyway

Reply:
Is there a way we could detect if the node is running, and report an error here? Just to protect against CLI accidents?

Reply:
Maybe we could check if a process was using the 26657 or 26656 ports

Reply:
If this has to connect to the database, this will fail anyway. See: 
https://github.com/tendermint/tendermint/pull/6785#pullrequestreview-737362127